### PR TITLE
Sync rna-transcription

### PR DESCRIPTION
**Pull request by fejan92**
Changes which have been made till now.

- change: uuid and testdox added in test file for rna-transcription
- MD file formatted for rna-transcription
- remove: strict types from rna-transcription

Needs to be done for 

Decide on adding / adjusting / ordering test cases to match current problem specs
metadata: unsynced: rna-transcription

https://github.com/exercism/php/issues/730

### CODE REVIEW

File: exercises/practice/rna-transcription/.docs/instructions.md [hunk 1]
```diff
@@ -15,6 +15,6 @@ Given a DNA strand, its transcribed RNA strand is formed by replacing each nucle
 - `T` -> `A`
 - `A` -> `U`
 
-~~~~exercism/note
+```exercism/note
```
Comment (mk-mxp) [inline exercises/practice/rna-transcription/.docs/instructions.md:18]: Doon't change the markdown files manually. Use `configlet` only.

File: exercises/practice/rna-transcription/.docs/introduction.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ You work for a bioengineering company that specializes in developing therapeutic
 
 Your team has just been given a new project to develop a targeted therapy for a rare type of cancer.
 
-~~~~exercism/note
+```exercism/note
```
Comment (mk-mxp) [inline exercises/practice/rna-transcription/.docs/introduction.md:7]: Same as before, do not edit the markdown files.

File: exercises/practice/rna-transcription/.meta/example.php [hunk 3]
```diff
@@ -1,29 +1,5 @@
...
- * This directive not only affects the type declarations of parameters, but also
- * a function's return type.
- *
- * For more info review the Concept on strict type checking in the PHP track
- * <link>.
- *
- * To disable strict typing, comment out the directive below.
- */
-
-declare(strict_types=1);
```
Comment (mk-mxp) [inline exercises/practice/rna-transcription/.meta/example.php:25]: We want to have `declare(strict_types=1);`, only the huge comment above it is useless in this file.

File: exercises/practice/rna-transcription/RnaTranscription.php [hunk 4]
```diff
@@ -1,29 +1,5 @@
 <?php
 
-/*
```
Comment (mk-mxp) [inline exercises/practice/rna-transcription/RnaTranscription.php:3]: In this file (the students stub file), we want to have this comment and the `declare(strict_types=1);` directive. In general, this file should not change at all.

File: exercises/practice/rna-transcription/RnaTranscriptionTest.php [hunk 5]
```diff
@@ -1,58 +1,66 @@
...
- * This means that the strictness of typing is configured on a per-file basis.
- * This directive not only affects the type declarations of parameters, but also
- * a function's return type.
- *
- * For more info review the Concept on strict type checking in the PHP track
- * <link>.
- *
- * To disable strict typing, comment out the directive below.
+/**
+ * @uuid 123e4567-e89b-12d3-a456-426614174000
```
Comment (mk-mxp) [inline exercises/practice/rna-transcription/RnaTranscriptionTest.php:4]: Why is this here?

File: exercises/practice/rna-transcription/RnaTranscriptionTest.php [hunk 6]
```diff
@@ -1,58 +1,66 @@
...
  */
-
-declare(strict_types=1);
```
Comment (mk-mxp) [inline exercises/practice/rna-transcription/RnaTranscriptionTest.php:25]: In the tests, we also want this directive. Remove the comment only.

File: exercises/practice/rna-transcription/RnaTranscriptionTest.php [hunk 7]
```diff
@@ -1,58 +1,66 @@
...
 class RnaTranscriptionTest extends PHPUnit\Framework\TestCase
 {
     public static function setUpBeforeClass(): void
     {
         require_once 'RnaTranscription.php';
     }
 
+    /**
+     * @testdox It transcribes guanine to cytosine
+     * @uuid 123e4567-e89b-12d3-a456-426614174001
```
Comment (mk-mxp) [inline exercises/practice/rna-transcription/RnaTranscriptionTest.php:15]: In all the doc blocks above the tests:

`@uuid` is no valid annotation. We use a comment `uuid` only. Please change that.

```suggestion
     * uuid 123e4567-e89b-12d3-a456-426614174001
```

**Review by mk-mxp — CHANGES_REQUESTED** [review]
Please use the UUIDs and descriptions from [problem specification `canonical-data.json`](https://github.com/exercism/problem-specifications/blob/main/exercises/rna-transcription/canonical-data.json).

Each test case has a `"uuid"` field to use with the `uuid` comment and a `"description"` field to use with `@testdox`. Also make sure, that `"input"` and `"expected"` are used in the test method, not other values.

In the end, please sort the tests according to the order in `canonical-data.json`, as this is the recommended test order.

Thanks for jumping in and helping!

File: exercises/practice/rna-transcription/.docs/instructions.md (hunk 1 above)
Comment (fejan92) [reply exercises/practice/rna-transcription/.docs/instructions.md:18]: Ok I will update this.

File: exercises/practice/rna-transcription/.meta/example.php (hunk 3 above)
Comment (fejan92) [reply exercises/practice/rna-transcription/.meta/example.php:25]: Got it. Thanks!

File: exercises/practice/rna-transcription/RnaTranscriptionTest.php (hunk 5 above)
Comment (fejan92) [reply exercises/practice/rna-transcription/RnaTranscriptionTest.php:4]: I've added uuid in before class and each function also. Will update this also.

Comment (mk-mxp) [reply exercises/practice/rna-transcription/RnaTranscriptionTest.php:4]: Take a look at this file: https://github.com/exercism/php/blob/main/exercises/practice/yacht/YachtTest.php

No UUID for the class, or setUpBeforeClass, only the tests are identified by UUIDs

**Comment by mk-mxp** [comment]
@fejan92 Are you still working on this? Click "Re-request review" next to my name in "Reviewers" when you are done.

> Needs to be done for
> 
> Decide on adding / adjusting / ordering test cases to match current problem specs

I took a look into the existing solutions. It is OK to add the missing test as specified in `canonical-data.json` https://github.com/exercism/problem-specifications/blob/main/exercises/rna-transcription/canonical-data.json The expected value you used is not OK.

**Comment by fejan92** [comment]
> canonical-data.json

I have update the test cases here. For `canonical-data.json` I need to clone that project and update the test cases there also.

File: exercises/practice/rna-transcription/RnaTranscriptionTest.php [hunk 8]
```diff
@@ -31,28 +9,93 @@ public static function setUpBeforeClass(): void
...
     }
 
+    /**
+     * @testdox It transcribes all occurrences in a given sequence
+     * uuid 123e4567-e89b-12d3-a456-426614174005
+     */
     public function testTranscribesAllOccurrencesOne(): void
     {
         $this->assertSame('UGCACCAGAAUU', toRna('ACGTGGTCTTAA'));
     }
```
Comment (mk-mxp) [inline exercises/practice/rna-transcription/RnaTranscriptionTest.php:55]: `canonical-data.json` (https://github.com/exercism/problem-specifications/blob/main/exercises/rna-transcription/canonical-data.json) is **the source of test data** we use here in PHP track. That file has all data we need: the UUID, the description and the test cases.

All we do is write that data into our test file:

```suggestion
    /**
     * @testdox RNA complement
     * uuid 79ed2757-f018-4f47-a1d7-34a559392dbf
     */
    public function testTranscribesAllOccurrencesOne(): void
    {
        $this->assertSame('UGCACCAGAAUU', toRna('ACGTGGTCTTAA'));
    }
```

Do this for all tests. Remove the tests, that are not in `canonical-data.json`. Do not add new tests.

**Review by mk-mxp — CHANGES_REQUESTED** [review]

**Comment by mk-mxp** [comment]
> Only `input` and `expected` remain to add in textdoc documentation
> 
> Please tell me if the below is proper documentation after that I will update in the test file
> 
> ```
>     /**
>      * @testdox RNA complement
>      * @input UGCACCAGAAUU
>      * @expected UCGGCCGUGAAAU
>      * uuid 79ed2757-f018-4f47-a1d7-34a559392dbf
>      */
>     public function testTranscribesAllOccurrencesOne(): void
>     {
>        ...body
>     }
> ```

No, thanks. Don't do that.

**Review by mk-mxp — APPROVED** [review]
Thanks a lot for helping with that! There was much to learn about the Exercism internals for you. Nice work!

**Comment by fejan92** [comment]
Thank you.

Hope I am not messing up things. Yes as a beginner I am making mistakes.

**Pull request merged**